packages/metadata-common/src/MetadataEditor.tsx

```diff
@@ -100,6 +101,21 @@ export class MetadataEditor extends ReactWidget {
         if (!this.name) {
           this.title.label = `New ${this.schemaDisplayName}`;
         }
+        // Find categories of all schema properties
+        this.categories = { _noCategory: [] };
+        for (const schemaProperty in this.schema) {
+          const cat =
+            this.schema[schemaProperty].uihints &&
+            this.schema[schemaProperty].uihints.category;
+          if (!cat) {
+            this.categories['_noCategory'].push(schemaProperty);
+            // eslint-disable-next-line no-prototype-builtins
+          } else if (this.categories.hasOwnProperty(cat)) {
```

Comment:
an alternative to disabling the [eslint check is to write is as](https://eslint.org/docs/rules/no-prototype-builtins):

```javascript
} else if (Object.prototype.hasOwnProperty.call(this.categories, cat)) {
```

Reply:
Thanks!